src/events/components/CalendarView/CalendarHoverCard.tsx

```diff
@@ -3,19 +3,30 @@ import { faCalendarAlt as CalendarAlt } from '@fortawesome/free-regular-svg-icon
 import { faCalendarCheck as CalendarCheck } from '@fortawesome/free-regular-svg-icons/faCalendarCheck';
 import { faMapMarkedAlt as MapMarked } from '@fortawesome/free-solid-svg-icons/faMapMarkedAlt';
 import { FontAwesomeIcon as Icon } from '@fortawesome/react-fontawesome';
-import { IEvent } from 'events/models/Event';
 import { DateTime } from 'luxon';
-import React from 'react';
+import React, { FC } from 'react';
+
+import { useSelector } from 'core/redux/hooks';
+import { State } from 'core/redux/Store';
+import { IEvent } from 'events/models/Event';
+import { attendanceEventSelectors } from 'events/slices/attendanceEvents';
+import { eventSelectors } from 'events/slices/events';
+
 import EventImage from '../EventImage';
 import style from './calendar.less';
 
-const CalendarHoverCard = ({ location, image, company_event, attendance_event, title, event_start }: IEvent) => {
-  const registrationStart =
-    attendance_event && DateTime.fromISO(attendance_event.registration_start).toFormat('d MMM HH:mm');
-  const eventStart = DateTime.fromISO(event_start).toFormat('d MMM HH:mm');
+interface IProps {
+  eventId: number;
+}
+
+const CalendarHoverCard: FC<IProps> = ({ eventId }) => {
+  const event = useSelector((state) => eventSelectors.selectById(state, eventId) as IEvent);
```

Comment:
`..eventSelectors.selectById(state, eventId) as IEvent`
Do you have to type cast it to event? Is it not inferred from the adapters' type?

Reply:
Hehe, doing a bit of a shortcut. I'm typecasting away the `undefined`, since this component can't be rendered unless the event is defined

Reply:
Ah, right! So CalendarHoverCard will only be rendered if the event exists, and thus we can certain that the selector will never return `undefined`?
Cause if not, the next line `const { location, images, title, start_date } = event;` will throw a TypeError.
```js
const { location, images, title, start_date } = undefined;
TypeError: undefined has no properties.
```

Reply:
Yes, they are used in the calendar, where an event in the calendar is only rendered if it is already in the store